evalml/tests/automl_tests/test_automl_search_classification.py

```diff
@@ -574,51 +574,41 @@ def test_automl_allowed_pipelines_init_allowed_both_specified_multi(mock_fit, mo
+@patch('evalml.pipelines.MulticlassClassificationPipeline.score')
+@patch('evalml.pipelines.MulticlassClassificationPipeline.fit')
 @patch('evalml.pipelines.BinaryClassificationPipeline.score')
 @patch('evalml.pipelines.BinaryClassificationPipeline.fit')
-def test_automl_allowed_pipelines_search(mock_fit, mock_score, dummy_binary_pipeline_class, X_y_binary):
-    X, y = X_y_binary
-    mock_score.return_value = {'Log Loss Binary': 1.0}
-
-    allowed_pipelines = [dummy_binary_pipeline_class({})]
-    start_iteration_callback = MagicMock()
-    automl = AutoMLSearch(X_train=X, y_train=y, problem_type='binary', max_iterations=2, start_iteration_callback=start_iteration_callback,
-                          allowed_pipelines=allowed_pipelines)
-    automl.search()
-
-    assert start_iteration_callback.call_count == 2
-    assert start_iteration_callback.call_args_list[0][0][0] == BinaryClassificationPipeline
-    assert start_iteration_callback.call_args_list[1][0][0] == dummy_binary_pipeline_class
-
-
-def test_automl_binary_nonlinear_pipeline_search(nonlinear_binary_pipeline_class, X_y_binary):
-    X, y = X_y_binary
-
-    allowed_pipelines = [nonlinear_binary_pipeline_class({})]
-    start_iteration_callback = MagicMock()
-    automl = AutoMLSearch(X_train=X, y_train=y, problem_type='binary', max_iterations=2,
-                          start_iteration_callback=start_iteration_callback,
-                          allowed_pipelines=allowed_pipelines, n_jobs=1)
-    automl.search()
-
-    assert start_iteration_callback.call_count == 2
-    assert start_iteration_callback.call_args_list[0][0][0] == BinaryClassificationPipeline
-    assert start_iteration_callback.call_args_list[1][0][0] == nonlinear_binary_pipeline_class
+def test_automl_allowed_pipelines_search(mock_binary_fit, mock_binary_score,
+                                         mock_multi_fit, mock_multi_score,
+                                         is_linear, problem_type,
+                                         dummy_binary_pipeline_class, nonlinear_binary_pipeline_class,
+                                         dummy_multiclass_pipeline_class, nonlinear_multiclass_pipeline_class,
+                                         X_y_binary, X_y_multi):
+    if problem_type == ProblemTypes.BINARY:
+        X, y = X_y_binary
+        mock_binary_score.return_value = {'Log Loss Binary': 1.0}
+        expected_mock_class = BinaryClassificationPipeline
+        pipeline_class = dummy_binary_pipeline_class if is_linear else nonlinear_binary_pipeline_class
+    else:
+        X, y = X_y_multi
+        mock_multi_score.return_value = {'Log Loss Multiclass': 1.0}
+        expected_mock_class = MulticlassClassificationPipeline
 
+        pipeline_class = dummy_multiclass_pipeline_class if is_linear else nonlinear_multiclass_pipeline_class
 
-def test_automl_multiclass_nonlinear_pipeline_search_more_iterations(nonlinear_multiclass_pipeline_class, X_y_multi):
```

Comment:
Kind of confused by what this test was testing (that multiclass with more iterations will produce the right pipelines..?), but combined with above :)

Reply:
Sounds good! Yeah looking at this now, I'm not sure either haha. It feels like it could be an automl algo test which we put in automl search before those concepts had separation. Idk!